### PR TITLE
fix unit test pipeline

### PR DESCRIPTION
**Pull request by vicilliar**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bug fix

* **What is the current behavior?** (You can also link to an open issue here)
unit test pipeline does not work

* **What is the new behavior (if this is a feature change)?**
unit test pipeline should start up vespa first

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
in progress

* **Related Python client changes** (link commit/PR here)

* **Related documentation changes** (link commit/PR here)

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Comment by vicilliar** [comment]
sample run: https://github.com/marqo-ai/marqo/actions/runs/7550233094/job/20555560247

failing tests are the same as the ones failing on local

**Comment by vicilliar** [comment]
Passing unit test run (test vespa client skipped for now): https://github.com/marqo-ai/marqo/actions/runs/7620815655/job/20756184039

File: src/marqo/tensor_search/configs.py [hunk 1]
```diff
@@ -31,4 +31,5 @@ def default_env_vars() -> dict:
         EnvVars.MARQO_MAX_NUMBER_OF_REPLICAS: 1,
         EnvVars.MARQO_DEFAULT_EF_SEARCH: 2000,
         EnvVars.MARQO_ENABLE_BATCH_APIS: "FALSE",
+        EnvVars.MARQO_MAX_ADD_DOCS_COUNT: 64
```
Comment (farshidz) [inline src/marqo/tensor_search/configs.py]: We don't have this in v2. The limit is currently hard-coded

File: .github/workflows/unit_test_200gb_CI.yml [hunk 2]
```diff
@@ -73,19 +73,50 @@ jobs:
...
+
+          # Delete the zip file
+          rm vespa_tester_app.zip
+          echo "Deleted vespa_tester_app.zip"
           
       - name: Run Unit Tests
         run: |
-          export LOCAL_OPENSEARCH_URL="https://localhost:9200"
+          # Define these for use by marqo
+          export VESPA_CONFIG_URL=http://localhost:19071
```
Comment (farshidz) [inline .github/workflows/unit_test_200gb_CI.yml:114]: I think these are defaults anyway, but agree better to be explicit and define them

File: src/marqo/tensor_search/enums.py [hunk 3]
```diff
@@ -58,6 +58,7 @@ class EnvVars:
     MARQO_MAX_DELETE_DOCS_COUNT = "MARQO_MAX_DELETE_DOCS_COUNT"
     MARQO_MAX_NUMBER_OF_REPLICAS = "MARQO_MAX_NUMBER_OF_REPLICAS"
     MARQO_DEFAULT_EF_SEARCH = "MARQO_DEFAULT_EF_SEARCH"
+    MARQO_MAX_ADD_DOCS_COUNT = "MARQO_MAX_ADD_DOCS_COUNT"
```
Comment (farshidz) [inline src/marqo/tensor_search/enums.py]: We don't have this in v2. The limit is currently hard-coded

File: tests/tensor_search/test_delete_documents.py [hunk 4]
```diff
@@ -1,271 +1,271 @@
...
 
-    def setUp(self) -> None:
-        self.endpoint = self.authorized_url
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
 
-        self.generic_header = {"Content-type": "application/json"}
-        self.index_name_1 = "my-test-index-1"
+        text_index_with_random_model = cls.unstructured_marqo_index_request(model=Model(name='random'))
```
Comment (farshidz) [inline tests/tensor_search/test_delete_documents.py]: To be clear, this is not an index. It's an index request

File: tests/tensor_search/test_delete_documents.py [hunk 5]
```diff
@@ -1,271 +1,271 @@
...
+            ], tensor_fields=["f1"])
+
+        count1_res = self.monitoring.get_index_stats_by_name(self.text_index_with_random_model).number_of_documents
+
+        marqo.tensor_search.tensor_search.delete_documents(config=self.config, index_name=self.text_index_with_random_model, doc_ids=["455", "at-at"])
+
+        count_post_delete = self.monitoring.get_index_stats_by_name(self.text_index_with_random_model).number_of_documents
+
         assert count_post_delete == count0_res
+        assert count1_res == count0_res + 2
```
Comment (farshidz) [inline tests/tensor_search/test_delete_documents.py]: No need to change these, but do note that we're replacing these with `self.assert` to get a better output when it fails

File: tests/tensor_search/test_get_stats.py [hunk 6]
```diff
@@ -14,16 +14,10 @@ class TestGetStats(MarqoTestCase):
...
-        except IndexNotFoundError as s:
-            pass
+        self.clear_index_by_name(self.index_name_1)
 
     def tearDown(self) -> None:
-        try:
-            tensor_search.delete_index(config=self.config, index_name=self.index_name_1)
-        except IndexNotFoundError as s:
-            pass
+        self.clear_index_by_name(self.index_name_1)
```
Comment (farshidz) [inline tests/tensor_search/test_get_stats.py]: `tensor_search.get_stats` has been replaced by `monitoring` module. These tests should be failing to run? I think they're redundant now and go (or move to monitoring tests, if there's anything here that isn't covered there)

**Review by farshidz — CHANGES_REQUESTED** [review]

File: .github/workflows/unit_test_200gb_CI.yml (hunk 2 above)
Comment (vicilliar) [reply .github/workflows/unit_test_200gb_CI.yml:114]: yep I just defined them to be safe.

File: tests/tensor_search/test_delete_documents.py (hunk 5 above)
Comment (vicilliar) [reply tests/tensor_search/test_delete_documents.py]: Got this! Will use `self.assert` in future tests

File: tests/tensor_search/test_get_stats.py (hunk 6 above)
Comment (vicilliar) [reply tests/tensor_search/test_get_stats.py]: These tests are skipped for now, I will check for redundancies and remove this file

File: src/marqo/tensor_search/configs.py (hunk 1 above)
Comment (vicilliar) [reply src/marqo/tensor_search/configs.py]: removed

File: src/marqo/tensor_search/enums.py (hunk 3 above)
Comment (vicilliar) [reply src/marqo/tensor_search/enums.py]: removed

File: tests/tensor_search/test_delete_documents.py (hunk 4 above)
Comment (vicilliar) [reply tests/tensor_search/test_delete_documents.py]: nice catch, updated variable name to be more clear!

**Comment by vicilliar** [comment]
Passing unit test run with 1 vespa client test skipped: https://github.com/marqo-ai/marqo/actions/runs/7656834030/job/20866138183

File: tests/tensor_search/test_delete_documents.py [hunk 7]
```diff
@@ -1,271 +1,271 @@
...
 
-    def setUp(self) -> None:
-        self.endpoint = self.authorized_url
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
 
-        self.generic_header = {"Content-type": "application/json"}
-        self.index_name_1 = "my-test-index-1"
+        text_index_with_random_model_request = cls.unstructured_marqo_index_request(model=Model(name='random'))
```
Comment (farshidz) [inline tests/tensor_search/test_delete_documents.py]: Need to test with at least one structured and one unstructured index. Use subtests to branch inside each test

**Review by farshidz — CHANGES_REQUESTED** [review]

File: tests/tensor_search/test_delete_documents.py (hunk 7 above)
Comment (vicilliar) [reply tests/tensor_search/test_delete_documents.py]: I decided to do this with 2 separate files, similar to how add_documents tests were made.

**Review by farshidz — APPROVED** [review]

**Pull request merged**